src/CentrifugeRouter.sol

```diff
@@ -2,20 +2,47 @@
 pragma solidity 0.8.21;
 
 import {Auth} from "src/Auth.sol";
-import {IERC7540} from "src/interfaces/IERC7540.sol";
+import {Multicall} from "src/Multicall.sol";
 import {SafeTransferLib} from "src/libraries/SafeTransferLib.sol";
+import {IERC7540} from "src/interfaces/IERC7540.sol";
+import {IERC20} from "src/interfaces/IERC20.sol";
 import {ICentrifugeRouter} from "src/interfaces/ICentrifugeRouter.sol";
+import {IPoolManager} from "src/interfaces/IPoolManager.sol";
+
+contract CentrifugeRouter is Auth, Multicall, ICentrifugeRouter {
+    address public poolManager;
 
-contract CentrifugeRouter is Auth, ICentrifugeRouter {
     /// @inheritdoc ICentrifugeRouter
     mapping(address user => mapping(address vault => uint256 amount)) public lockedRequests;
 
+    constructor(address poolManager_) {
+        poolManager = poolManager_;
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
     // --- Administration ---
     /// @inheritdoc ICentrifugeRouter
     function recoverTokens(address token, address to, uint256 amount) external auth {
         SafeTransferLib.safeTransfer(token, to, amount);
     }
 
+    /// @inheritdoc ICentrifugeRouter
+    function file(bytes32 what, address data) external auth {
```

Comment:
This is missing a `File` event

Reply:
Updated!

Reply:
The event should be defined in the interface though :)

Reply:
Fixed!